lib/FreeRTOS/freertos-plus-mqtt/freertos_mqtt_agent.h

```diff
@@ -115,112 +99,180 @@ typedef struct CommandContext CommandContext_t;
+                             TransportInterface_t * pTransportInterface,
+                             MQTTGetCurrentTimeFunc_t getCurrentTimeMs,
+                             IncomingPublishCallback_t incomingCallback,
+                             void * pIncomingPacketContext );
+
 /**
  * @brief Process commands from the command queue in a loop.
  *
- * This demo requires a process loop command to be enqueued before calling this
- * function, and will re-add a process loop command every time one is processed.
- * This demo will exit the loop after receiving an unsubscribe operation.
+ * @param[in] pMqttAgentContext The MQTT agent to use.
  *
- * @return pointer to MQTT context that caused error, or `NULL` if terminated
- * gracefully.
+ * @return appropriate error code, or `MQTTSuccess` from a successful disconnect
+ * or termination.
  */
-MQTTContext_t * MQTTAgent_CommandLoop( void );
+MQTTStatus_t MQTTAgent_CommandLoop( MQTTAgentContext_t * pMqttAgentContext );
 
 /**
  * @brief Resume a session by resending publishes if a session is present in
- * the broker, or reestablish subscriptions if not.
+ * the broker, or clear state information if not.
  *
- * @param[in] mqttContextHandle Handle to the MQTT connection to resume.
+ * @param[in] pMqttAgentContext The MQTT agent to use.
  * @param[in] sessionPresent The session present flag from the broker.
  *
  * @return `MQTTSuccess` if it succeeds in resending publishes, else an
  * appropriate error code from `MQTT_Publish()`
  */
-MQTTStatus_t MQTTAgent_ResumeSession( MQTTContextHandle_t mqttContextHandle,
+MQTTStatus_t MQTTAgent_ResumeSession( MQTTAgentContext_t * pMqttAgentContext,
                                       bool sessionPresent );
 
 /**
  * @brief Add a command to call MQTT_Subscribe() for an MQTT connection.
  *
  * @param[in] mqttContextHandle Handle to the MQTT connection to use.
  * @param[in] pSubscriptionInfo Struct describing topic to subscribe to.
- * @param[in] incomingPublishCallback Incoming publish callback for the subscriptions.
- * @param[in] incomingPublishCallbackContext Context for the publish callback.
- * @param[in] commandCompleteCallback Optional callback to invoke when the command completes.
- * @param[in] commandCompleteCallbackContext Optional completion callback context.
- * @param[in] blockTimeMS The maximum amount of time in milliseconds to wait for the
- * command to be posted to the MQTT agent should the MQTT agent's event queue be
- * full.  Tasks wait in the Blocked state so don't use any CPU time.
+ * @param[in] pCommandInfo The information pertaining to the command, including:
+ *  - cmdCompleteCallback Optional callback to invoke when the command completes.
+ *  - pCmdCompleteCallbackContext Optional completion callback context.
+ *  - blockTimeMs The maximum amount of time in milliseconds to wait for the
+ *    command to be posted to the MQTT agent should the MQTT agent's event queue
```

Comment:
```suggestion
 *    command to be posted to the MQTT agent, should the MQTT agent's event queue
```

Reply:
I don't believe a comma is necessary in this sentence but it doesn't hurt